x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -79,105 +102,329 @@ export class AlertsClient {
-            },
-          },
-        },
+        // @ts-expect-error
+        body: queryBody,
         seq_no_primary_term: true,
       });
 
-      if (result == null || result.body == null || result.body.hits.hits.length === 0) {
-        return;
+      if (!result?.body.hits.hits.every((hit) => isValidAlert(hit._source))) {
+        const errorMessage = `Invalid alert found with id of "${id}" or with query "${query}" and operation ${operation}`;
+        this.logger.error(errorMessage);
+        throw Boom.badData(errorMessage);
       }
 
-      if (!isValidAlert(result.body.hits.hits[0]._source)) {
-        const errorMessage = `Unable to retrieve alert details for alert with id of "${id}".`;
-        this.logger.debug(errorMessage);
-        throw new Error(errorMessage);
-      }
+      await this.ensureAllAuthorized(result.body.hits.hits, operation);
 
-      return {
-        ...result.body.hits.hits[0]._source,
-        _version: encodeHitVersion(result.body.hits.hits[0]),
-      };
+      result?.body.hits.hits.map((item) =>
+        this.auditLogger?.log(
+          alertAuditEvent({
+            action: operationAlertAuditActionMap[operation],
+            id: item._id,
+            outcome: 'unknown',
+          })
+        )
+      );
+
+      return result.body;
     } catch (error) {
-      const errorMessage = `Unable to retrieve alert with id of "${id}".`;
-      this.logger.debug(errorMessage);
-      throw error;
+      const errorMessage = `Unable to retrieve alert details for alert with id of "${id}" or with query "${query}" and operation ${operation} \nError: ${error}`;
+      this.logger.error(errorMessage);
+      throw Boom.notFound(errorMessage);
     }
   }
 
-  public async get({
-    id,
-    index,
-  }: GetAlertParams): Promise<ParsedTechnicalFields | null | undefined> {
+  /**
+   * When an update by ids is requested, do a multi-get, ensure authz and audit alerts, then execute bulk update
+   * @param param0
+   * @returns
+   */
+  private async mgetAlertsAuditOperate({
+    ids,
+    status,
+    indexName,
+    operation,
+  }: {
+    ids: string[];
+    status: STATUS_VALUES;
+    indexName: string;
+    operation: ReadOperations.Find | ReadOperations.Get | WriteOperations.Update;
+  }) {
     try {
-      // first search for the alert by id, then use the alert info to check if user has access to it
-      const alert = await this.fetchAlert({
-        id,
-        index,
+      const mgetRes = await this.esClient.mget<ParsedTechnicalFields>({
+        index: indexName,
+        body: {
+          ids,
+        },
       });
 
-      if (alert == null) {
-        return;
+      await this.ensureAllAuthorized(mgetRes.body.docs, operation);
+
+      for (const id of ids) {
+        this.auditLogger?.log(
+          alertAuditEvent({
+            action: operationAlertAuditActionMap[operation],
+            id,
+            ...(operation === WriteOperations.Update ? { outcome: 'unknown' } : { operation }),
```

Comment:
You don't need to pass `operation` in if this is not a write operation:
```suggestion
            ...(operation === WriteOperations.Update ? { outcome: 'unknown' } : {}),
```

Reply:
Cleaned this up here: 36838ab2e10ba9b2640665345758373075e0fa44